great_expectations/expectations/metrics/metric_provider.py

```diff
@@ -76,6 +77,78 @@ def __new__(cls, clsname, bases, attrs):
         return newclass
 
 
+# The following is based on "https://stackoverflow.com/questions/9008444/how-to-warn-about-class-name-deprecation".
+class DeprecatedMetaMetricProvider(MetaMetricProvider):
```

Comment:
Have we dealt with deprecation before and if so, can you point me to that transition? Do we need to make a `MetricProvider`-specific deprecation class or can we make one general one and use it for all instances in the future? This seems like a great deal of complexity for something that should have a reasonably simple solution.

I don't love adding additional dependencies but what are your thoughts on implementing a simple decorator like this: https://deprecation.readthedocs.io/en/latest/

I really like marking the version so when we generate our API docs, it is very explicit as to when and why we deprecated.

Reply:
This deprecation is a simple rename but what if we're deprecating because we're adding new functionality? Does the solution you've written best fit that scenario?

Reply:
This works specifically for all subclasses of `MetricProvider` -- we can adopt a similar technique for the others, but there would be slight differences, which is completely reasonable.

We have a lot of use of the deprecation at the feature and method level (e.g., the occurrences of `DeprecationWarning`) throughout.  Thanks.

Reply:
Per our conversation, I'm okay with this implementation if it is the only way to work at the metaclass level. I think I'd be more comfortable if someone more knowledgeable about deprecation and metaclasses took a quick look. I would like to revisit to discuss deprecation and logging strategy down the line so I've made a ticket in our Jira backlog.

In terms of code changes, could you please place this class definition below `MetricProvider` or move it to a new file? I don't want to overshadow the primary focus of this file, which is the base class. Also, updating the docstring a bit to explain usage and what it does when a deprecated class is initialized/called would be helpful.

Reply:
@cdkini I think your point about discussing "how" we do deprecation is a good one... in this case, I may be missing some context on the "larger picture" surrounding this rename, so I'm sorry if this is too simple of a question. 

What would be the harm of adding functionality to the `ColumnMetricProvider` directly? (without doing the rename).  Does the concern have to do with the increase in "scope" of what the MetricProvider class does, with the problem being `ColumnMetricProvider` referring to single-columns, when now it can also handle `ColumnAggregateMetrics`?.